pkg/services/datasources/models.go

```diff
@@ -147,6 +147,18 @@ type DeleteDataSourceCommand struct {
 	UpdateSecretFn UpdateSecretFn
 }
 
+// DeleteDataSourceSecrets will delete the SecureJsonData on DataSource based on OrgID as well as the UID (preferred), ID, or Name.
+// At least one of the UID, ID, or Name properties must be set in addition to OrgID.
+type DeleteDataSourceSecretsCommand struct {
+	ID   int64
+	UID  string
+	Name string
+
+	OrgID int64
```

Comment:
Could optimize this to only allow ID and potentially in addition OrgID to A) speed up the datasource operation and B) since there's only one consumer of this does make sense to optimize for that consumer rather than potential future requirements.

Reply:
Given my other https://github.com/grafana/grafana/pull/50463#discussion_r913141904 this command might no longer be needed

Reply:
Will remove this in favor of the changes suggested in https://github.com/grafana/grafana/pull/50463#discussion_r913141904